src/system/WatchableEventManagerSelect.h

```diff
@@ -41,9 +41,9 @@ class WatchableSocket;
 class WatchableEventManager
 {
 public:
-    void Init(System::Layer & systemLayer);
-    void Shutdown();
-    void Signal();
+    CHIP_ERROR Init(System::Layer & systemLayer);
+    CHIP_ERROR Shutdown();
+    CHIP_ERROR Signal();
```

Comment:
Can you document these?

When can signalling fail?

Reply:
I'll make `Signal()` succeed or die.

`Init()` and `Shutdown()` may return implementation-dependent errors, just like the `System::Layer` versions. `WatchableEventManager` wants to be `SystemLayerImpl` when it grows up.